app/models/application_form.rb

```diff
@@ -314,8 +311,31 @@ def enough_references_have_been_provided?
     application_references.feedback_provided.count >= MINIMUM_COMPLETE_REFERENCES
   end
 
+  def address_formatted_for_geocoding
+    full_address.compact.join(', ')
+  end
+
 private
 
+  def geocode_address_if_required
+    return unless address_changed?
+
+    if international?
+      update!(latitude: nil, longitude: nil)
+    else
+      GeocodeApplicationAddressWorker.perform_async(id)
```

Comment:
Out of interest, why not just pass the application_form in?

Reply:
Sidekiq [recommends](https://github.com/mperham/sidekiq/wiki/Best-Practices#1-make-your-job-parameters-small-and-simple) keeping job parameters simple. The docs encourage using simple identifiers rather than saving state.

Reply:
Okay thanks for the link. I'm just going to test it out locally, but the code looks good 👍